### PR TITLE
minor update in README example code

### PR DESCRIPTION
**Pull request by zachcheung**
*  fix error: `no new variables on left side of :=`
* print dump filename instead of filename format
* add mkdir automatically

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,12 +45,11 @@ func main() {
...
 
   // Dump database to file
-  err := dumper.Dump()
-  if err != nil {
+  if err := dumper.Dump(); err != nil {
     fmt.Println("Error dumping:", err)
     return
   }
-  fmt.Printf("File is saved to %s", dumpFilenameFormat)
+  fmt.Printf("File is saved to %s\n", dumper.Out.(*os.File).Name())
```
Comment (BrandonRoehl) [inline README.md]: If we are casting to an underlying type can we check if it truly is a part of that interface?

```go
if file, ok := dumper.Out.(*os.File); ok {
  fmt.Println("File is saved to", file.Name())
}
```

**Review by BrandonRoehl — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (zachcheung) [reply README.md]: yep, `dumper.Out.(*os.File)` should be OK here, if not, would be failed in previous dumper register, but I'll update according to your suggestion  if you think it must be checked.

Comment (BrandonRoehl) [reply README.md]: Correct. I just worry about people copying things directly out of the README. `doc.go` contains the same snippet as well. I'd be fine with either removing `doc.go` or updating it with the new README information

Comment (zachcheung) [reply README.md]: You're right, just updated it.

**Review by BrandonRoehl — APPROVED** [review]

**Comment by BrandonRoehl** [comment]
Looks good I’ll have to run the changes by the rest of the team tomorrow before merge

**Comment by BrandonRoehl** [comment]
Just noticed the rendering at https://github.com/ZachCheung/go-mysqldump and it looks like the version of markdown that github is using needs the marks

```
\```
go
```

**Review by brazeaudavid — APPROVED** [review]

**Review by BrandonRoehl — CHANGES_REQUESTED** [review]
Just noticed the rendering at https://github.com/ZachCheung/go-mysqldump and it looks like the version of markdown that github is using needs the marks

```
\```go
```

**Review by BrandonRoehl — APPROVED** [review]

**Pull request merged**